tests/routes/artifact_test.py

```diff
@@ -97,3 +90,17 @@ def test_illegal_artifact_upload(self):
             .route_params(bucket_name="test", path="_test") \
             .expect(403, self.RESPONSE_INVALID_NAME) \
             .post(data={"file": (StringIO("file contents"), "test.txt")}, headers=self.auth)
+
+    def test_bucket_configuration_doesnt_exist(self):
+        self.route_tester \
+            .artifact() \
+            .route_params(bucket_name="lol-this-doesnt-exist", path="hello/there") \
+            .expect(404, self.RESPONSE_404) \
+            .get(headers=self.auth)
+
+    def test_bucket_doesnt_exist(self):
+        self.route_tester \
+            .artifact() \
+            .route_params(bucket_name="thisBucketDoesntExistLol", path="hello/there") \
+            .expect(500, self.response_500()) \
+            .get(headers=self.auth)
```

Comment:
If we configuration doesn't exist that (at least partially) means that we explicitly do not want to deal with a bucket.  From an api perspective, we say it doesn't exist.

However, if we explicitly configured a bucket, but we fail to connect to it, or it doesn't exist, there is nothing a user can do about that.

I don't understand the part about mapped to RESOURCE_NOT_FOUND.   We explicitly map BucketNotFoundError to 500